public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -89,15 +96,28 @@ export class ApiKeysPageUnconnected extends PureComponent<Props, State> {
 
   onMigrateApiKeys = async () => {
     try {
-      this.onMigrateAll();
-      let serviceAccountsUrl = '/org/serviceaccounts';
-      locationService.push(serviceAccountsUrl);
-      window.location.reload();
+      const response = await this.props.migrateAll();
+      this.setState((prevState: State) => {
+        return {
+          ...prevState,
+          operationSummaryVisible: true,
+          operationSummaryData: response,
+        };
+      });
     } catch (err) {
       console.error(err);
     }
   };
 
+  dismissModal = async () => {
+    this.setState((prevState: State) => {
```

Comment:
This might be simplier
```suggestion
    this.setState({ operationSummaryVisible: false });
```

Reply:
do i not have to include the prevstate?